package net.eliasaa.tutorialmod.item;

import net.eliasaa.tutorialmod.Tutorialod;
import net.eliasaa.tutorialmod.item.custom.ChiselItem;
import net.fabricmc.fabric.api.creativetab.v1.CreativeModeTabEvents;
import net.minecraft.core.Registry;
import net.minecraft.core.registries.BuiltInRegistries;
import net.minecraft.core.registries.Registries;
import net.minecraft.resources.Identifier;
import net.minecraft.resources.ResourceKey;
import net.minecraft.world.item.CreativeModeTabs;
import net.minecraft.world.item.Item;

import java.util.function.Function;

public class modItems {
        public static final Item FLUORITE = registerItem("fluorite", Item::new);
        public static final Item RAW_FLUORITE = registerItem("raw_fluorite", Item::new);

        public static final Item CHISEL = registerItem("chisel", properties -> new ChiselItem(properties.durability(64)));


    private static Item registerItem(String name, Function<Item.Properties, Item> function) {
        return Registry.register(BuiltInRegistries.ITEM, Identifier.fromNamespaceAndPath(Tutorialod.MOD_ID, name),
                function.apply(new Item.Properties().setId(ResourceKey.create(Registries.ITEM, Identifier.fromNamespaceAndPath(Tutorialod.MOD_ID, name)))));

    }


    public static void registerModItems(){
        Tutorialod.LOGGER.info("Registrering Mod For Items" + Tutorialod.MOD_ID);

        CreativeModeTabEvents.modifyOutputEvent(CreativeModeTabs.INGREDIENTS).register(output -> {
            output.accept(FLUORITE);
            output.accept(RAW_FLUORITE);
        });
    }
}
